Automattic/WP-Job-Manager
# Add bulk update check

Part of #2417 

### Changes proposed in this Pull Request

* Switch over to the bulk update endpoint.
* Remove old tests and rewrite based on new techniques. I'll add additional tests as I migrate things to the new endpoints.
* Fixed issue that prevented plugin installed multiple times from getting updates.

### Testing instructions

* Make sure your local WPJMCOM is on the `add/icon-info` branch.
* Use the following snippet to point WPJM to your local env:
```
add_filter( 'pre_http_request', function( $preempt, $r, $url ) {
	$host = parse_url( $url, PHP_URL_HOST );

	if ( 'wpjobmanager.com' === $host ) {
		$scheme  = parse_url( $url, PHP_URL_SCHEME );
		$http    = _wp_http_get_object();
		$new_url = str_replace(
			$scheme . '://' . $host,
			'http://wpjobmanagercom.vipdev.lndo.site',
			$url
		);

		error_log( 'Redirect from: ' . $url );
		error_log( 'To: ' . $new_url );

		return $http->request( $new_url, $r );
	}
	return $preempt;
}, 10, 3 );
add_filter( 'http_request_host_is_external', function($external) {
	return true;
});
```
- Install older versions of some WP Job Manager paid add-ons. Clear your site transient cache (`wp transient delete --all --network`).
- Go to Job Listings > All Jobs and make sure the updates available notice shows up.
- With licenses deactivated, ensure plugin updates show up for those plugins in WP Admin > Dashboard > Updates. No downloads should appear.
- Ensure notice still shows with plugin update available from WP Admin > Job Listings.
* Activate a license.
* Ensure plugin updates show up as available. 
* Install plugin update.
* Make sure it removes the plugin from the notice at the top of WP Admin > Job Listings. 

- 

### Deprecated Code

* `WP_Job_Manager_Helper_API::plugin_update_check`
* `WP_Job_Manager_Helper::get_installed_plugins` signature changed. Soft requirement for new second argument `$keyed_by_filename`.
